File: tyama_codeeval80.rb
#!/usr/bin/ruby
require 'cgi'
while gets
a,b=$_.chomp.split(';').map{|e|CGI.unescape(e)+'/'}
a=~/^(.*?):\/\/(.*?)\/(.*)$/
c=[$1,$2,$3]
b=~/^(.*?):\/\/(.*?)\/(.*)$/
d=[$1,$2,$3]
if !c[1].index(':') then c[1]+=':80' end
if !d[1].index(':') then d[1]+=':80' end
if c[1].index(':')==c[1].size-1 then c[1]+='80' end
if d[1].index(':')==d[1].size-1 then d[1]+='80' end
puts c[0].casecmp(d[0])==0 && c[1].casecmp(d[1])==0 && c[2]==d[2] ?'True':'False'
end

File: codeeval/tyama_codeeval80.pl
#!/usr/bin/perl
use strict;
use warnings;
sub url_decode{
	my $str = shift;
	$str =~ tr/+/ /;
	$str =~ s/%([0-9A-Fa-f][0-9A-Fa-f])/pack('H2', $1)/eg;
	return $str;
}
while(<>){
	chomp;
	my ($a,$b)=map {url_decode($_).'/'} split(';');
	$a=~/^(.*?):\/\/(.*?)\/(.*)$/;
	my @c=($1,$2,$3);
	$b=~/^(.*?):\/\/(.*?)\/(.*)$/;
	my @d=($1,$2,$3);
	$c[1].=':80' if(index($c[1],':')<0);
	$d[1].=':80' if(index($d[1],':')<0);
	$c[1].='80' if(index($c[1],':')==length($c[1])-1);
	$d[1].='80' if(index($d[1],':')==length($d[1])-1);
	print lc($c[0]) eq lc($d[0]) && lc($c[1]) eq lc($d[1]) && $c[2] eq $d[2] ? "True\n" : "False\n";
}